### PR TITLE
refactor: exprience change classes to attributes

### PR DESCRIPTION
**Pull request by supproduction**
renames tag
changes classes to attributes
uses signals
improves component logic

closes [HRZ-3037](https://spryker.atlassian.net/browse/HRZ-3037) [HRZ-3036](https://spryker.atlassian.net/browse/HRZ-3036) [HRZ-3038](https://spryker.atlassian.net/browse/HRZ-3038)

[HRZ-3037]: https://spryker.atlassian.net/browse/HRZ-3037?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HRZ-3036]: https://spryker.atlassian.net/browse/HRZ-3036?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HRZ-3038]: https://spryker.atlassian.net/browse/HRZ-3038?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/platform/experience/composition/src/composition.component.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    const layoutStyles = this.layoutStyles();
+    const inlineStyles = this.layoutBuilder.collectStyles(components);
+
+    return html`${repeat(
+      components,
+      (component) => component.id,
+      (component, index) => this.renderComponent(component, index)
+    )}
+    ${when(layoutStyles, () => unsafeHTML(`<style>${layoutStyles}</style>`))}
+    ${when(inlineStyles, () => unsafeHTML(`<style>${inlineStyles}</style>`))} `;
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/composition/src/composition.component.ts]: can we end up having a single style element when there's layout and/or inline styles?

File: libs/platform/experience/composition/src/composition.component.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  protected $uidFromRoute = effect(() => {
+    if (!this.route) {
+      return;
+    }
+
+    this.uid = (
+      signal(this.experienceService.getComponent({ route: this.route }))() ?? {}
+    ).id;
+  });
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/composition/src/composition.component.ts:45]: can we move this inside the `$component` computed?

File: libs/platform/experience/composition/src/composition-preview.component.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    const actions = ['mouseover', 'click', 'mouseout', 'add'];
+    const interaction = this.$interaction();
+
+    if (!actions.includes(interaction?.action)) {
+      return;
+    }
+
+    const focusedNameAttr = 'name';
+    const root = this.shadowRoot;
+    const focusedComponent = root?.querySelector(`.${EB_PREVIEW_FOCUS_CLASS}`);
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/composition/src/composition-preview.component.ts]: we could use `@query('.eb-preview-focus')` instead

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/platform/experience/composition/src/composition.component.ts (hunk 2 above)
Comment (supproduction) [reply libs/platform/experience/composition/src/composition.component.ts:45]: Not sure, in this case we recalculate uid inside component (which is depended from uid) which might lead to circular issue

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**